### PR TITLE
Added Dashboard route and dashboard in the sidebar

### PR DESCRIPTION
**Pull request by abhishek-01k**
## Pull Request Template

#1600 

### Description

Added the Dashboard route and added dashboard button in the sidebar.

- **Problem/Feature**:

### Type of Change

- [x] New feature

### Frontend Guidelines

- [x] Followed frontend guidelines as per the [Guidelines Notion Doc](https://www.notion.so/pushprotocol/Frontend-dApp-Guidelines-1d7806ae3d9e4569a340b563dcd0536c)

### Build & Testing

- [x] No errors in the build terminal
- [x] Engineer has tested the changes on their local environment
- [ ] Engineer has tested the changes on deploy preview

### Review & Approvals

- [x] Self-review completed
- [ ] Code review by at least one other engineer
- [ ] Documentation updates if applicable

### Notes

Things that are done:

- [x] Proper folder structuring is done. However some sub-folders or files are not present feel free to add them according to the usage.

Things that are left in this PR:

- [x] New sidebar design according to the figma is not done yet. Colors might mismatch because that are left for V2. In Figma, we have new color designs.
- [x] Logo of the Dashboard is not imported from Blocks because all the sidebar logos are imported from a single file and code is written in that way to access that file. So, Will work on this too.
- [ ] Layouts for the right side is not done in this PR. I will be doing it in the #1601  issue's PR.

What the reviewer needs to do:

- Check the functionality of the sidebar buttons are working or not. Proper route is coming or not.
- Also check the sidebar in the mobile view.
- UI of the sidebar.
- see if the sidebar is not breaking on any viewport or not.

### CODE REVIEW

File: src/components/MobileNavButton.jsx [hunk 1]
```diff
@@ -40,11 +40,20 @@ function MobileNavButton({ item, data, sectionID, active, bg = 'none', showNavBa
   let definedMargin;
 
   switch (sectionID) {
+
     case GLOBALS.CONSTANTS.NAVBAR_SECTIONS.MOBILE:
       SelectedIcon = LeftBarPrimaryItemIcon;
       definedMargin = '5px';
       break;
-    case GLOBALS.CONSTANTS.NAVBAR_SECTIONS.SECONDARY:
```
Comment (rohitmalhotra1420) [inline src/components/MobileNavButton.jsx:47]: Why have we done these changes? Is this a previously missing edge case?

**Review by mishramonalisha76 — CHANGES_REQUESTED** [review]
Need to change the folder name of Dashboard to lower case.
Rest everything look good

File: src/structure/MobileNavigation.tsx [hunk 2]
```diff
@@ -51,27 +51,27 @@ function MobileNavigation({ showNavBar, setShowNavBar }) {
 
     let newNavSetup = navigationSetup;
     if (processingState === null) {
-      newNavSetup.secondary[0].data.hidden = true;
-      newNavSetup.secondary[1].data.hidden = true;
+      newNavSetup.developersList[0].data.hidden = true;
+      newNavSetup.developersList[1].data.hidden = true;
```
Comment (rohitmalhotra1420) [inline src/structure/MobileNavigation.tsx:55]: What does this change do?

File: src/structure/Navigation.tsx [hunk 3]
```diff
@@ -54,27 +54,27 @@ function Navigation() {
 
     let newNavSetup = navigationSetup;
     if (processingState === null) {
-      newNavSetup.secondary[0].data.hidden = true;
-      newNavSetup.secondary[1].data.hidden = true;
+      newNavSetup.developersList[0].data.hidden = true;
```
Comment (rohitmalhotra1420) [inline src/structure/Navigation.tsx:57]: Why?

**Review by rohitmalhotra1420 — CHANGES_REQUESTED** [review]

File: src/config/NavigationList.js [hunk 4]
```diff
@@ -33,136 +33,154 @@ const NavigationList = {
...
+        iconFactory: null,
+        name: 'Chat',
         title: 'Chat',
-        light: {
-          bg: GLOBALS.COLORS.GRADIENT_SECONDARY,
-          fg: themeLight.headerTagFg,
-        },
-        dark: {
-          bg: themeDark.headerTagBg,
-          fg: themeDark.headerTagFg,
```
[the remark this answers is not in the document]
Comment (abhishek-01k) [reply src/config/NavigationList.js:83]: Its not deleted its present

File: src/modules/Dashboard/Dashboard.tsx [hunk 5]
```diff
@@ -1,11 +1,13 @@
-import React from 'react';
+import { FC } from 'react';
```
Comment (rohitmalhotra1420) [inline src/modules/Dashboard/Dashboard.tsx:1]: @abhishek-01k folder name still not fixed. It needs to be modules/dashboard/Dashboard.tsx

**Review by rohitmalhotra1420 — CHANGES_REQUESTED** [review]

**Review by rohitmalhotra1420 — APPROVED** [review]

**Pull request merged**